### ISSUE TITLE
[2.0.1] Undefined index: balias

### ISSUE THREAD

**Issue by Naoto-Ida** [phalcon/cphalcon#10321]
I have been having problems with using the `executeQuery()` method of `modelsManager`.

I get the error below:
```
Notice: Undefined index: balias in XXXController.php on line XXX
Warning: Phalcon\Mvc\Model\Resultset\Complex::current(): Property should be string in XXXController.php on line X
```

Here is my code:
```
$sql = "SELECT * from namespace1\table1 inner join namespace2\table1 on namespace1\table1.id = namespace2\table1.id";
$result = $this->modelsManager->executeQuery($sql);
print_r($result->toArray());
```

I've read through several issues regarding namespaces and join statements,
but am not sure if it's a bug with either one of them,
or my misuse of the method, so thought I'd report it.

But I am guessing it is a bug, since I have never tried to access an index of `balias` in this code.

Thanks in advance.

**Comment by SidRoberts** [comment]
As you can probably tell, I've had a look at this. :smile:

The problem is likely this: https://github.com/phalcon/cphalcon/blob/2.0.x/phalcon/mvc/model/resultset/complex.zep#L232

Tbh, I'm slightly confused by the `current()` method and I'd rather not make a PR without fully understanding the implications it might have.

For others attempting to fix this:

* The `attribute` variable seems to be used for two separate things.
* See https://github.com/phalcon/cphalcon/blob/1.3.4/ext/mvc/model/resultset/complex.c#L335
* I did originally add https://github.com/phalcon/cphalcon/blob/1.3.4/ext/mvc/model/resultset/complex.c#L372 as well but it [failed during testing](https://travis-ci.org/phalcon/cphalcon/builds/62209870).
* I don't think `attribute` should allowed to be `null` on these lines:
 * https://github.com/phalcon/cphalcon/blob/2.0.x/phalcon/mvc/model/resultset/complex.zep#L261
 * https://github.com/phalcon/cphalcon/blob/2.0.x/phalcon/mvc/model/resultset/complex.zep#L265

Once this bug is fixed, I'll look at simplifying the method.

**Comment by Naoto-Ida** [comment]
@SidRoberts Hey, thanks!
So should I wait for it to be fixed, or maybe downgrade to 1.3.4?

**Comment by Naoto-Ida** [comment]
Nvm, gonna try to fiddle with the file before compiling it, then see how it works. Thanks!

**Comment by andresgutierrez** [comment]
Fixed in the 2.0.x branch
